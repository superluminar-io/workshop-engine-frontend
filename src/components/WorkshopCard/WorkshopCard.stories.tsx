import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { WorkshopCard } from "./WorkshopCard";
import { WorkshopMenu } from "../WorkshopMenu/WorkshopMenu";
import EditIcon from "@mui/icons-material/Edit";
import { MenuItem } from "@mui/material";

export default {
  title: "WorkshopCard",
  component: WorkshopCard,
} as ComponentMeta<typeof WorkshopCard>;

const TemplateSingle: ComponentStory<typeof WorkshopCard> = (args) => (
  <WorkshopCard {...args} />
);

const TemplateList: ComponentStory<typeof WorkshopCard> = (args) => (
  <>
    <WorkshopCard {...args} />
    <WorkshopCard {...args} />
    <WorkshopCard {...args} />
  </>
);

export const Single = TemplateSingle.bind({});
Single.storyName = "single item";
Single.args = {
  title: "Serverless Beginner Workshop",
  awsAccountId: "12345678",
};

export const WithMenu = TemplateSingle.bind({});
WithMenu.storyName = "w/ menu";
WithMenu.args = {
  title: "Serverless Beginner Workshop",
  awsAccountId: "12345678",
  headerRightSide: (
    <WorkshopMenu id="abc">
      <MenuItem>
        <EditIcon />
        Edit
      </MenuItem>
    </WorkshopMenu>
  ),
};

export const WithoutAwsAccountId = TemplateSingle.bind({});
WithoutAwsAccountId.storyName = "single item w/o AWS Account ID";
WithoutAwsAccountId.args = {
  title: "Serverless Beginner Workshop",
};

export const Loading = TemplateSingle.bind({});
Loading.storyName = "loading";
Loading.args = {
  variant: "loading",
};

export const List = TemplateList.bind({});
List.storyName = "list";
List.args = {
  title: "Serverless Beginner Workshop",
  awsAccountId: "12345678",
};
