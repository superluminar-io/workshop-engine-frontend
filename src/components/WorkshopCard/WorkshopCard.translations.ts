export const buttonLabel = "Open AWS Management Console";
