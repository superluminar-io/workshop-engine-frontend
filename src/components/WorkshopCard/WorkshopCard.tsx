import React from "react";
import {
  Button,
  Card,
  CardContent,
  CardActions,
  Typography,
} from "@mui/material";

export interface WorkshopCardProps {
  title: string;
  awsAccountId: string;
}

export const WorkshopCard: React.FunctionComponent<WorkshopCardProps> = ({
  title,
  awsAccountId,
}) => {
  return (
    <Card sx={{ mb: 4 }}>
      <CardContent>
        <Typography variant="h5" component="span">
          {title}
        </Typography>
      </CardContent>
      <CardActions>
        <Button
          href={`/api/console?awsAccountId=${awsAccountId}`}
          target="_blank"
        >
          Open AWS Management Console
        </Button>
      </CardActions>
    </Card>
  );
};
