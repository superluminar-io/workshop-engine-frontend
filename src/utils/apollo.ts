import { ApolloClient, createHttpLink, InMemoryCache } from '@apollo/client';
import { setContext } from '@apollo/client/link/context';
import { useSession } from '@clerk/nextjs';

const httpLink = createHttpLink({
  uri: process.env.NEXT_PUBLIC_APPSYNC_URL,
});

const authLink = setContext(async (_, { headers }) => {
  const { session: { getToken } } = useSession();
  const token = await getToken();
  
  return {
    headers: {
      ...headers,
      authorization: token ? `Bearer ${token}` : "",
    }
  }
});

export const client = new ApolloClient({
  link: authLink.concat(httpLink),
  cache: new InMemoryCache()
});
