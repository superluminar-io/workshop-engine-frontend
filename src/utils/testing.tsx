import React from "react";
import {
  render as renderTestingLibrary,
  RenderOptions,
  RenderResult,
} from "@testing-library/react";
import { ThemeProvider } from "@mui/material";
import { theme } from "../config/theme";

export const noop = () => null;

export const render = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => {
  return renderTestingLibrary(
    <ThemeProvider theme={theme}>{ui}</ThemeProvider>,
    options
  );
};
