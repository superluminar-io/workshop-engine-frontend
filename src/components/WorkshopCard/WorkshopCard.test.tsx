import { render } from "../../utils/testing";
import { WorkshopCard } from "./WorkshopCard";

it("renders the layout", () => {
  const { container } = render(
    <WorkshopCard title="Title" awsAccountId="12345678" />
  );
  expect(container).toMatchSnapshot();
});
