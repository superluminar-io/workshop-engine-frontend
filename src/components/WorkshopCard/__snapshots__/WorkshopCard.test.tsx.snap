// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the layout 1`] = `
.emotion-0 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 12px;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 0px 1px;
  overflow: hidden;
  margin-bottom: 32px;
}

.emotion-1 {
  padding: 16px;
}

.emotion-1:last-child {
  padding-bottom: 24px;
}

.emotion-2 {
  margin: 0;
  font-family: "Source Sans Pro",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 8px;
}

.emotion-3>:not(:first-of-type) {
  margin-left: 8px;
}

.emotion-4 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Source Sans Pro",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.75;
  text-transform: uppercase;
  min-width: 64px;
  padding: 6px 8px;
  border-radius: 12px;
  -webkit-transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,box-shadow 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,border-color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,color 250ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  color: #ff3b61;
}

.emotion-4::-moz-focus-inner {
  border-style: none;
}

.emotion-4.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-4 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(255, 59, 97, 0.04);
}

@media (hover: none) {
  .emotion-4:hover {
    background-color: transparent;
  }
}

.emotion-4.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-0"
  >
    <div
      class="MuiCardContent-root emotion-1"
    >
      <span
        class="MuiTypography-root MuiTypography-h5 emotion-2"
      >
        Title
      </span>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing emotion-3"
    >
      <a
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  emotion-4"
        href="/api/console?awsAccountId=12345678"
        tabindex="0"
        target="_blank"
      >
        Open AWS Management Console
      </a>
    </div>
  </div>
</div>
`;
